### DISCUSSION TITLE
Cannot connect to HomeKit

### DISCUSSION THREAD

**Discussion by Tortel007** [HomeSpan/HomeSpan#768] [Q&A]
Hey guys, 
this is my first time using GitHub and the HomeSpan library. I have issues to get the first example running. I uploaded the code to the esp32 and typed "W" in the console to connect to WiFi. I can see the esp32 in my router. When searching for new devices in HomeKit the device doesn't show up. Is there anything I do wrong?
Thanks for your help, have a great day :)

**Comment by HomeSpan** [comment]
What do you mean by searching for new devices in HomeKit?  Do you mean you are trying to pair a new Accessory (the plus button in the Home App) but the Home App does not show the HomeSpan device as an option for pairing?

**Reply by Tortel007** [reply]
Yes, correct. It worked some time ago with the example code. I've changed nothing and now the accessory does not show up when pairing. I tried different smartphones and WiFis but nothing happens.

**Reply by HomeSpan** [reply]
If the device has previously been paired it cannot be paired a second time unless it is first unpaired.  It won't even show up in the Home App if you try to find it when pairing a second time.  

When you start up the device, what does it show in the Serial Monitor - paired or not yet paired?

If the device says it is currently paired, you can reset the state to unpaired by typing 'H' into the Serial Monitor.  This erases all pairing info as well as resets the ID of the device so it looks fresh from the perspective of the Home App and should now be pairable again. 

Note the device and the Home App can get out of sync like this if you previously unpaired the device a while back from the Home App but the device itself was not powered on and did not get the unpair command from the Home App.  The same can happen if you start the pairing process and cancel before finished.

**Reply by Tortel007** [reply]
Thank you for your help. The device was still paired and I could reset it with 'H'. Again, thank you very much!

**Comment by denizsahin7744** [comment]
You need to connect same wifi with esp32 otherwise you can't see in home app
